navbar.ts

```diff
@@ -0,0 +1,433 @@
+      if (template.getAttribute("data-type") === "back") {
+        const newContent = template.content.cloneNode(true) as DocumentFragment;
+        this.backElement = document.createElement("a");
+        this.backElement.appendChild(newContent.cloneNode(true));
+      } else if (template.getAttribute("data-type") === "close") {
+        const newContent = template.content.cloneNode(true) as DocumentFragment;
+        this.closeElement = document.createElement("a");
+        this.closeElement.appendChild(newContent.cloneNode(true));
+      }
+    });
+
+    this.backElement.id = "back";
+    this.closeElement.id = "close";
+    this.parentContainer["menu"] = this.noParentContianer;
+    if (this.cutoffwidth === 0) {
+      this.cutoffwidth = this.getNavbarLength();
+    }
+
+    window.addEventListener("resize", this.updateNavbar.bind(this));
+    this.updateNavbar();
+
+    this.menuElement.onclick = this.handleMenu.bind(this);
+    this.closeElement.onclick = this.handleClose.bind(this);
+
+    this.menu = document.createElement("div");
+    if (this.mode == "overlay") {
+      this.menu.appendChild(this.overlayNode());
+    } else if (this.mode == "saccordion") {
+      this.menu.appendChild(this.accordionCreator());
+    } else {
+      this.menu.appendChild(this.accordionCreator("menu", false));
+    }
+    this.menu.style.display = "none";
+    document.body.appendChild(this.menu);
+
+    for (const key in this.overlayContainer) {
+      const anchor = this.getAnchorElement(key);
+      if (anchor) {
+        anchor.onclick = () => {
+          if (this.currentPath.length > 0) {
+            const previousAnchor =
+              this.currentPath[this.currentPath.length - 1];
+
+            const attributes = this.parseAttributes(
+              previousAnchor.getAttribute("attr") || ""
+            );
+            this.overlayContainer[attributes.nm][0].style.display = "none";
+          }
+          this.overlayContainer[key][0].style.display = "block";
+          this.currentPath.push(anchor);
+        };
+      }
+    }
+
+    const elements = document.querySelectorAll("#close");
+    elements.forEach((element) => {
+      (element as HTMLAnchorElement).onclick = () => this.handleClose();
+    });
+
+    const ele = document.querySelectorAll("#back");
+    ele.forEach((element) => {
+      (element as HTMLAnchorElement).onclick = () => this.handleBack();
+    });
+
+    this.menu.classList.add(this.canvascls);
+  }
+
+  overlayNode(parentName: string = "menu"): HTMLElement {
+    const dive = document.createElement("div");
+    if (parentName == "menu") {
+      const div = document.createElement("div");
+      div.appendChild(this.closeElement);
+      dive.appendChild(div);
+    }
+    this.parentContainer[parentName].forEach((child) => {
+      const div = document.createElement("div");
+      div.appendChild(child);
+      dive.appendChild(div);
+      const attributes = this.parseAttributes(child.getAttribute("attr") || "");

```

Comment:
We seem to be parsing the attributes in multiple places, Dont you think we can save this somewhere in the container object.